subsys/bluetooth/audio/shell/bap.c

```diff
@@ -2330,6 +2349,149 @@ static struct bt_le_scan_cb bap_scan_cb = {
 #if defined(CONFIG_BT_AUDIO_RX)
 static unsigned long recv_stats_interval = 100U;
 
+#if defined(CONFIG_LIBLC3)
+struct lc3_data {
+	void *fifo_reserved; /* 1st word reserved for use by FIFO */
+	struct net_buf *buf;
+	struct shell_stream *sh_stream;
+};
+
+K_MEM_SLAB_DEFINE(lc3_data_slab, sizeof(struct lc3_data), CONFIG_BT_ISO_RX_BUF_COUNT,
+		  __alignof__(struct lc3_data));
+
+static int16_t lc3_rx_buf[LC3_MAX_NUM_SAMPLES];
+static K_FIFO_DEFINE(lc3_in_fifo);
+
+static int init_lc3_decoder(struct shell_stream *sh_stream)
+{
+	if (sh_stream == NULL) {
+		shell_error(ctx_shell, "invalid stream to init LC3");
+		return -EINVAL;
+	}
+
+	if (sh_stream->lc3_decoder != NULL) {
+		shell_error(ctx_shell, "Already initialized");
+		return -EALREADY;
+	}
+
+	if (sh_stream->lc3_freq_hz == 0 || sh_stream->lc3_frame_duration_us == 0) {
+		shell_error(ctx_shell, "Invalid freq (%u) or frame duration (%u)",
+			    sh_stream->lc3_freq_hz, sh_stream->lc3_frame_duration_us);
+
+		return -EINVAL;
+	}
+
+	shell_print(ctx_shell,
+		    "Initializing the LC3 decoder with %u us duration and %u Hz frequency",
+		    sh_stream->lc3_frame_duration_us, sh_stream->lc3_freq_hz);
+	/* Create the decoder instance. This shall complete before stream_started() is called. */
+	sh_stream->lc3_decoder = lc3_setup_decoder(sh_stream->lc3_frame_duration_us,
+						   sh_stream->lc3_freq_hz, 0, /* No resampling */
```

Comment:
I am not sure if the bap shell intended for non-USB device? 

if not only for non-USB device, the sameple rate should not always 0.

Reply:
This PR does not touch USB, but only does decoding. My followup PRs add proper resampling for USB